From e261ede97ae49ec1edd3cf53d4e74886cae58288 Mon Sep 17 00:00:00 2001
From: Hien Huynh <hien.huynh.px@renesas.com>
Date: Tue, 27 Jun 2023 15:10:23 +0700
Subject: [PATCH] makefile.linaro: update makefile to be built with
 cross-compiler

Update all cross-compiler environment assignment from "=" to "?="
to let system use the environment variable from toolchain if they
are set.

Signed-off-by: Hien Huynh <hien.huynh.px@renesas.com>
---
 makefile.linaro | 14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

diff --git a/makefile.linaro b/makefile.linaro
index 6dd9a53..a49cfc1 100755
--- a/makefile.linaro
+++ b/makefile.linaro
@@ -85,17 +85,17 @@ DEPEND_FILE = $(patsubst %.lib, ,$(OBJ_FILE:%.o=%.d))
 
 ###################################################
 #C compiler
-CC = $(CROSS_COMPILE)gcc
+CC ?= $(CROSS_COMPILE)gcc
 #Assembler
-AS = $(CROSS_COMPILE)as
+AS ?= $(CROSS_COMPILE)as
 #Linker
-LD = $(CROSS_COMPILE)ld
+LD ?= $(CROSS_COMPILE)ld
 #Liblary
-AR = $(CROSS_COMPILE)ar
+AR ?= $(CROSS_COMPILE)ar
 #Object dump
-OBJDMP = $(CROSS_COMPILE)objdump
+OBJDUMP ?= $(CROSS_COMPILE)objdump
 #Object copy
-OBJCOPY = $(CROSS_COMPILE)objcopy
+OBJCOPY ?= $(CROSS_COMPILE)objcopy
 
 #PYTHON
 EXE_PYTHON = python3
@@ -153,7 +153,7 @@ $(OUTPUT_FILE): $(OBJ_FILE_BOOT) $(OBJ_FILE) $(MEMORY_DEF)
 
 
 #   Dis assemble
-	$(OBJDMP) -d -S "$(OUTPUT_FILE)" > "$(FILE_NAME)_disasm.txt"
+	$(OBJDUMP) -d -S "$(OUTPUT_FILE)" > "$(FILE_NAME)_disasm.txt"
 	
 #	Time Stamp
 	@echo ==========  %date% %time%  ==========
-- 
2.25.1

